### ISSUE TITLE
ogr2ogr throwing error while generating shape file

### ISSUE THREAD

**Issue by Niraj1411** [OSGeo/gdal#1511]
## Expected behavior and actual behavior.

While generating the shape file using the below command"
ogr2ogr -f "ESRI Shapefile" export_05022019.shp OCI:<username>/<password>@<host_server>:<port_number>/<db_name> -SQL "SELECT A ,B ,NVL(C,1)  AS C FROM <table_name>"

I am getting the error as:

Warning 1: Value '1.0745' of field SELECT A ,B ,NVL(C,1)  AS C FROM <table_name>.C parsed incompletely to integer 1.

Without NVL function the shape file is getting  generated successfully.

## Operating system

Windows 64 bit

## GDAL version and provenance

Using GDAL version 2

**Comment by rouault** [comment]
you should probably add an explicit cast around NVL(C,1) AS C to force it to float. It looks like the data type currently guessed is integer

**Comment by jratike80** [comment]
Have you tried with "SELECT A ,B ,NVL(C,1.0000)... ?

**Comment by Niraj1411** [comment]
> you should probably add an explicit cast around NVL(C,1) AS C to force it to float. It looks like the data type currently guessed is integer

Thanks a lot. By casting explicitly to float, I am not getting error. But now the new issue is, the generated value is upto 6 digits only after the decimal point, whereas I have values with up to 10 or 11 digits after the decimal point. Not sure if this is the default behavior of this command to generate the shape file.

My command looks like this:  
ogr2ogr -f "ESRI Shapefile" export_05022019.shp OCI:/@<host_server>:<port_number>/<db_name> -SQL "SELECT A ,B ,CAST(NVL(C,1) AS FLOAT) AS C FROM <table_name>"

**Comment by jratike80** [comment]
If it is enough to get the right result then you could create the shapefile without NVL and update nulls afterwards into what you want with` ogrinfo -sql "UPDATE layer set..."`
